cmd/editorconfig-checker/main.go

```diff
@@ -18,8 +18,8 @@ import (
 // version is dynamically set at compiletime
 var version string
 
-// defaultConfigFilePath determines where the config is located
-const defaultConfigFilePath = ".ecrc"
+// defaultConfigFileNames determines the file names where the config is located
+var defaultConfigFileNames = []string{".ecrc", ".editorconfig-checker.json"}
```

Comment:
Shouldn't it be the opposite?


```suggestion
var defaultConfigFileNames = []string{".editorconfig-checker.json", ".ecrc"}
```

Reply:
Performance wise, it doesn't change much. And if we first check `.ecrc`, we optimize for existing users, if we first check `.editorconfig-checker.json`, we optimize for users migrating to the new config file name.

Reply:
Objection! 😅😁😂

![OBJECTION!.jpg](https://github.com/user-attachments/assets/7070f878-c5b0-4c48-b83c-fb12479c2eb8)

The performance argument is not acceptable.

The file will be accessed once in a software that literally opens several, if not almost all,  files in a repository. And read all their content.

So no, a file not found hit is not a problem.

For me, the question is clearly functional.

Anyone starting using the new filename must have the settings contained in it, even there is .ecrc file in the folder

Reply:
> The performance argument is not acceptable.
> 
> The file will be accessed once in a software that literally opens several, if not almost all, files in a repository. And read all their content.
> 
> So no, a file not found hit is not a problem.

Yes, I'm not against this change, but I don't understand why it matters, was thinking about "performance", but as we both agree, it pretty much doesn't change anything. 

We might have a deprecation notice for `.ecrc`. But we don't support merging of configuration files anyway, so the users should not use both `.ecrc` and `.editorconfig-checker.json`, so we might also warn the user, that only 1 config file is taken. Anyway, we will be able to simplify the code again, to only support `.editorconfig-checker.json` in the next major version, so it doesn't seems worth it to "work on it too much".

Anyway, I applied your suggestion.